internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -1124,10 +1186,73 @@ func (cmi *ClusterManager) updateMVList(rvMap map[string]dcache.RawVolume, exist
 				break
 			}
 		}
+
+		// Call join mv and check if all rv's are able to join the mv with rv list
+		deleteRv, err := cmi.joinMV(mvName, existingMVMap[mvName])
+		if err != nil {
+			// If RPC server has not started we delete the new mv added and remove the node from the nodeToRvs map
+			// and redo distribution
+			// TODO :: Would prefer if we could validate this behaviour before allocating RVs to the Mv
```

Comment:
Since we need to send all the component RVs in JoinMV, we have to do it after all RVs are assigned.. 
we can start with this.. mostly JoinMV will not fail so we should be fine, perf wise